### PR TITLE
Two list mode functions to get and set position in list mode data stream

### PR DESCRIPTION
**Pull request by robbietuk**
Added functions: `get_list_mode_position()` and `set_list_mode_position(pos)`

`get_list_mode_position()` returns an unsigned long with the current position in the list mode data

`set_list_mode_position(pos)` set the current position to `pos`

These are two functions I have developed for a _"block"_ subset sampling method for list mode.

### CODE REVIEW

**Comment by robbietuk** [comment]
As per @rijobro 's advice, I have modified the old method to `set_get_position()` to depend on `set_list_mode_position()` to remove code duplication

**Comment by robbietuk** [comment]
I might have made a mess of this now. I tried to reset the branch and cherrypick the changes I made.

File: src/include/stir/listmode/CListModeData.h [hunk 1]
```diff
@@ -31,6 +31,7 @@
 #include "stir/ProjDataInfo.h"
 #include "stir/IO/ExamData.h"
 #include "stir/RegisteredParsingObject.h"
+#include "stir/Succeeded.h"
```
Comment (rijobro) [inline src/include/stir/listmode/CListModeData.h]: You shouldn't need this include now (we put it in during debugging)

File: src/listmode_buildblock/CListModeDataECAT8_32bit.cxx [hunk 2]
```diff
@@ -136,5 +136,20 @@ set_get_position(const CListModeDataECAT8_32bit::SavedPosition& pos)
...
+{
+    return current_lm_data_ptr->get_list_mode_position();
+}
+
+Succeeded
+CListModeDataECAT8_32bit::
+set_list_mode_position(const unsigned long pos)
+{
+    return current_lm_data_ptr->set_list_mode_position(pos);
+    //return Succeeded::no;
```
Comment (rijobro) [inline src/listmode_buildblock/CListModeDataECAT8_32bit.cxx]: can delete this commented line

**Comment by KrisThielemans** [comment]
I really don't see the point of this. What does "position" mean in a generic listmode file? People might be tempted to think it counts records, but it won't. Even for file formats with same size of record, it'll fail in cases where there's a header. 

Once you say "position is just an arbitrary marker to allow be to go back", it's exactly the same as the `save_get_position` in my opinion.

As this is a change that breaks backwards compatibility (classes that are not currently in STIR would need extra stuff), I'm reluctant.

let's talk about it.

**Comment by robbietuk** [comment]
After discussing this with Kris, I agree that this functionality is not generally compatible with general list mode file formats. Possibly `get_list_mode_position()` has some use in debugging but without its set counterpart might be redundant. 

I will keep both methods in my own branch. Shall I close this pull request?

**Pull request closed without merging**